### ISSUE TITLE
Consider using <Link> instead of <a> in main-menu.tsx for non-externals

### ISSUE THREAD

**Issue by peterbe** [mdn/yari#872]
Just a note to remember to play with this.

**Comment by peterbe** [comment]
The task is actually to decide if client-side navigation is better than regular navigation. 
It's a delicate quest that has one foot in web performance and one foot in a11y. 
Whatever we decide, if the evidence is good, we should consider this for the other places where we use `<Link>` which is the breadcrumbs and the autocomplete search results.

**Comment by peterbe** [comment]
@schalkneethling Given the potential improvements to the `<Link>` with the [`<PreloadingDocumentLink>` component](https://github.com/mdn/yari/pull/1593) we might reconsider. 
We'd need to be aware of this for each link in the header. I.e. whether it links to a document or to some other page that doesn't have a `index.json` state file.

**Comment by Ryuno-Ki** [comment]
How does `<Link>` behave with context actions (save Link As…, Open Link in New Tab etc)?
Can I drag a link to the URL bar and have it opened?

Since you mentioned accessibility (a11y for short), here are the relevant sections from WAI-ARIA:
https://www.w3.org/TR/wai-aria-practices/#link and https://www.w3.org/TR/wai-aria-1.1/#link

**Comment by caugner** [comment]
Closing as "won't fix", as we have mostly migrated away from `<Link />`.
